internal/db/arango.go

```diff
@@ -118,14 +140,96 @@ func (c *ArangoClient) QueryAnnotations(ctx context.Context, key string) ([]docu
 
 	var annotations []documents.Annotation
 	for {
-		var doc documents.Annotation
-		_, err := cursor.ReadDocument(ctx, &doc)
+		_, err := cursor.ReadDocument(ctx, &annotations)
 		if driver.IsNoMoreDocuments(err) {
 			break
 		} else if err != nil {
 			return nil, err
 		}
-		annotations = append(annotations, doc)
 	}
+
 	return annotations, nil
 }
+
+func (c *ArangoClient) QueryScoreByLayer(ctx context.Context, key string, layer contracts.LayerType) ([]documents.Score, error) {
+	db, err := c.instance.Database(ctx, c.cfg.DatabaseName)
+	if err != nil {
+		return nil, err
+	}
+
+	var query string
+	switch layer {
+	case contracts.Application:
+		query = `FOR s IN scores FILTER s.dataRef == @key AND s.layer == @layer RETURN [s]`
+	case contracts.CiCd:
+		query = `FOR appScore IN scores FILTER appScore.dataRef == @key 
+				LET cicdScore = (
+					FOR s IN scores FILTER 
+					s.layer == @layer AND s.tag ANY IN appScore.tag 
```

Comment:
There can exist multiple score nodes for cicd layer, we should sort by their timestamp and limit the selection to one score.
```suggestion
					s.layer == @layer AND s.tag ANY IN appScore.tag SORT s.timestamp DESC LIMIT 1 
```

Reply:
@michaelehab I am assuming this is due to failed pipelines, or multiple pipelines producing the same binary/image. I am wondering though does this affect having a piece of data passing through multiple workloads deployed from different pipeliens ?